aiida_mlip/calculations/base.py

```diff
@@ -206,42 +226,10 @@ def prepare_for_submission(
 
         # Define architecture from model if model is given,
         # otherwise get architecture from inputs and download default model
-        architecture = None
-        architecture = (
-            str((self.inputs.model).architecture)
-            if "model" in self.inputs and hasattr(self.inputs.model, "architecture")
-            else str(self.inputs.arch.value) if "arch" in self.inputs else None
-        )
-
-        if architecture:
-            cmd_line["arch"] = architecture
-
-        model_path = None
-        if "model" in self.inputs:
-            # Raise error if model is None
-            if self.inputs.model is None:
-                raise ValueError("Model cannot be None")
-            model_path = self.inputs.model.filepath
-        else:
-            if "config" in self.inputs and "model" in self.inputs.config:
-                model_path = None
-            else:
-                if "arch" in self.inputs:
-                    # if model is not given (which is different than it being None)
-                    model_path = ModelData.download(
-                        "https://github.com/stfc/janus-core/raw/main/tests/models/mace_mp_small.model",  # pylint: disable=line-too-long
-                        architecture,
-                    ).filepath
-        if model_path:
-            cmd_line.setdefault("calc-kwargs", {})["model"] = model_path
+        cmd_line = self._add_arch_to_cmdline(cmd_line)
+        cmd_line = self._add_model_to_cmdline(cmd_line)
```

Comment:
i am a little bit confused here... the model seems to overwrite the arch... should not be additive?